### PR TITLE
feat: Add `Remove share targets` patch

### PR DESCRIPTION
**Pull request by 1fexd**
This patch fully disables direct sharing targets for a selected app (highlighted in the screenshot [here](https://developer.android.com/training/sharing/direct-share-targets#publish-direct-share-targets)) which might be unwanted (e.g. social media direct messages), since these can not be disabled or configured by the user

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/all/shortcut/removetarget/RemoveShareTargetsPatch.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+@Patch(
+    name = "Remove all share targets",
+    description = "Removes share targets like directly sharing to a frequent contact of an app",
+    use = false,
+)
+@Suppress("unused")
+object RemoveShareTargetsPatch : ResourcePatch() {
+    override fun execute(context: ResourceContext) {
+        val shortcutsXml = context["res/xml/shortcuts.xml", false]
+        if (!shortcutsXml.exists()) return
```
[the remark this answers is not in the document]
Comment (1fexd) [reply src/main/kotlin/app/revanced/patches/all/shortcut/removetarget/RemoveShareTargetsPatch.kt:18]: Definitely sounds like a good idea, what do you think @oSumAtrIX

[the remark this answers is not in the document]
Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/all/shortcut/removetarget/RemoveShareTargetsPatch.kt:18]: You can use the logger java.logging.Logger, it's just that no patch has used it yet.

[the remark this answers is not in the document]
Comment (1fexd) [reply src/main/kotlin/app/revanced/patches/all/shortcut/removetarget/RemoveShareTargetsPatch.kt:18]: Hm, creating an instance via `Logger#getLogger(String)` doesn't seem to print to the patch output, at least not when using the cli

[the remark this answers is not in the document]
Comment (1fexd) [reply src/main/kotlin/app/revanced/patches/all/shortcut/removetarget/RemoveShareTargetsPatch.kt:18]: Would you mind giving an example? I tried instantiating it via `Logger.getLogger(RemoveShareTargetsPatch::class.java.simpleName)` which works, but doesn't print anything

[the remark this answers is not in the document]
Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/all/shortcut/removetarget/RemoveShareTargetsPatch.kt]: You have to use the full name: https://github.com/ReVanced/revanced-patches/pull/3334/files#diff-41e0d997ab5b000a8e8f1dc192caf7b0c7923f886be4a8223aa4cbb414f1e810R23

**Pull request merged**